crates/iceberg/src/spec/view_metadata.rs

```diff
@@ -0,0 +1,675 @@
+pub struct ViewMetadataBuilder(ViewMetadata);
+
+impl ViewMetadataBuilder {
+    /// Creates a new view metadata builder from the given view metadata.
+    pub fn new(origin: ViewMetadata) -> Self {
+        Self(origin)
+    }
+
+    /// Creates a new view metadata builder from the given view creation.
+    pub fn from_view_creation(view_creation: ViewCreation) -> Result<Self> {
+        let ViewCreation {
+            location,
+            schema,
+            properties,
+            name: _,
+            representations,
+            default_catalog,
+            default_namespace,
+            summary,
+        } = view_creation;
+        let initial_version_id = super::INITIAL_SEQUENCE_NUMBER;
+        let version = ViewVersion::builder()
+            .with_default_catalog(default_catalog)
+            .with_default_namespace(default_namespace)
+            .with_representations(representations)
+            .with_schema_id(schema.schema_id())
+            .with_summary(summary)
+            .with_timestamp_ms(Utc::now().timestamp_millis())
+            .with_version_id(initial_version_id)
+            .build();
+
+        let versions = HashMap::from_iter(vec![(initial_version_id, version.into())]);
+
+        let view_metadata = ViewMetadata {
+            format_version: ViewFormatVersion::V1,
+            view_uuid: Uuid::now_v7(),
+            location,
+            current_version_id: initial_version_id,
+            versions,
+            version_log: Vec::new(),
+            schemas: HashMap::from_iter(vec![(schema.schema_id(), Arc::new(schema))]),
+            properties,
+        };
+
+        Ok(Self(view_metadata))
+    }
+
+    /// Changes uuid of view metadata.
+    pub fn assign_uuid(mut self, uuid: Uuid) -> Result<Self> {
+        self.0.view_uuid = uuid;
+        Ok(self)
+    }
+
+    /// Returns the new view metadata after changes.
+    pub fn build(self) -> Result<ViewMetadata> {
+        Ok(self.0)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+#[serde(rename_all = "kebab-case")]
+/// A log of when each snapshot was made.
+pub struct ViewVersionLog {
+    /// ID that current-version-id was set to
+    version_id: i64,
+    /// Timestamp when the view's current-version-id was updated (ms from epoch)
+    timestamp_ms: i64,
+}
+
+impl ViewVersionLog {
+    #[inline]
+    /// Creates a new view version log.
+    pub fn new(version_id: i64, timestamp: i64) -> Self {
+        Self {
+            version_id,
+            timestamp_ms: timestamp,
+        }
+    }
+
+    /// Returns the version id.
+    #[inline]
+    pub fn version_id(&self) -> i64 {
+        self.version_id
+    }
+
+    /// Returns the timestamp in milliseconds from epoch.
+    #[inline]
+    pub fn timestamp_ms(&self) -> i64 {
+        self.timestamp_ms
+    }
+
+    /// Returns the last updated timestamp as a DateTime<Utc> with millisecond precision.
+    pub fn timestamp(self) -> MappedLocalTime<DateTime<Utc>> {
```

Comment:
Conversion to `DateTime<Utc>` is fallible - thus it would become a `Result`. We already have `timestamp_ms()` as a non-fallible version. Do you still think we should change?

Reply:
What I absolutely don't like is the `TableMetadata` way of doing this which just panics:
https://github.com/apache/iceberg-rust/blob/f5b406f08f85d8bca402ce840962deee9d6b34c0/crates/iceberg/src/spec/table_metadata.rs#L148

Reply:
https://github.com/c-thiel/iceberg-rust/commit/d354e63e912cf059f7234c1a0a77f3c3dfb7d219

Reply:
And my proposal for `TableMetadata`.
https://github.com/c-thiel/iceberg-rust/commit/1c49ff91e68f53c769e1d3df2c1aff8ab6d348a5

Do you have a better Idea for method naming? I think `last_updated_ms` was a really bad method for returning a `DateTime` in the first place. Nonetheless, it now changes the behavior to return an int instead so that we still have a non-fallible accessor.